clash-lib/src/Clash/Normalize/Strategy.hs

```diff
@@ -56,6 +56,7 @@ normalization =
     xOptim     = bottomupR (apply "xOptimize" xOptimize)
     cleanup    = topdownR (apply "etaExpandSyn" etaExpandSyn) >->
                  topdownSucR (apply "inlineCleanup" inlineCleanup) !->
+                 topdownSucR (apply "inlineIdentities" inlineIdentity) >->
```

Comment:
I think you want to change these around. Since your implementation of `inlineCleanup` doesn't actually inline the let-binders: it replaces the RHS of a let-binding by a proper call to `unsafeCoerce#`. `inlineCleanup` will do the actual inlining. It's probably worth adding a comment w.r.t. to the above.

Reply:
right, I'm aware that I do stretch the semantics of 'inline' here a lot.
what's a good name? `collapseRHSNoops` ?

Reply:
Yeah, that sounds good. I'd still like to see a comment along the lines of:
```haskell
{- [Note] relation `collapseRHSNoops` and `inlineCleanup`

The `collapseRHSNoops` transformation replaces functions/primitives that are the identity in HDL, 
but not in Haskell, by `unsafeCoerce`. `inlineCleanup` subsequently inlines these `unsafeCoerce` calls.
The end result of all of this is that we get no/fewer assignments in HDL where the RHS is simply a 
variable reference.

See issue #779
-}
```
in the haddock comments for `collapseRHSNoops` function and then do:
```haskell
-- See [Note] relation `collapseRHSNoops` and `inlineCleanup`
topdownSucR (apply "collapseRHSNoops" collapseRHSNoops) >->
topdownSucR (apply "inlineCleanup" inlineIdentity) !->
```